pkg/lite-apiserver/proxy/proxy.go

```diff
@@ -306,37 +283,58 @@ func (p *EdgeReverseProxy) writeCache(header http.Header, statusCode int, pipeRe
 	}
 	bodyBytes, err := holder.Output()
 	if err != nil {
-		klog.Errorf("Write cache marshal %s error: %v", p.key(), err)
+		klog.Errorf("Write cache marshal %s error: %v", key, err)
 		return err
 	}
-	err = p.storage.Store(p.key(), bodyBytes)
+	err = p.storage.Store(key, bodyBytes)
 	if err != nil {
-		klog.Errorf("Write cache %s error: %v", p.key(), err)
+		klog.Errorf("Write cache %s error: %v", key, err)
 		return err
 	}
 
 	return nil
 }
 
-func getRequestProperties(r *http.Request) (isList bool, isWatch bool, needCache bool) {
+func getRequestVerb(r *http.Request) (isList bool, isWatch bool) {
```

Comment:
Beter define a new function to get request info. And use that in both `getRequestVerb` and `needCache`.